dc-cudami-editor/src/components/modals/ImageAdderModal.jsx

```diff
@@ -0,0 +1,161 @@
+import ImageRenderingHintsForm from './imageAdder/ImageRenderingHintsForm'
+import ImageUploadForm from './imageAdder/ImageUploadForm'
+import ImageUrlForm from './imageAdder/ImageUrlForm'
+
+class ImageAdderModal extends Component {
+  constructor(props) {
+    super(props)
+    this.initialAttributes = {
+      alignment: 'left',
+      altText: '',
+      caption: '',
+      label: '',
+      linkNewTab: true,
+      linkUrl: '',
+      title: '',
+      url: '',
+      width: '33%',
+    }
+    this.state = {
+      attributes: this.initialAttributes,
+      readOnly: false,
+      step: 'initial',
+    }
+    subscribe('editor.show-image-modal', () => {
+      this.props.onToggle()
+    })
+  }
+
+  addImageToEditor = () => {
+    this.toggle()
+    const filteredAttributes = Object.entries(this.state.attributes).reduce(
+      (filteredAttributes, attribute) => {
+        if (attribute[1]) {
+          filteredAttributes[attribute[0]] = attribute[1]
+        }
+        return filteredAttributes
+      },
+      {}
+    )
+    publish('editor.add-image', filteredAttributes)
+  }
+
+  getStepComponent = t => {
+    const STEP_COMPONENT_MAPPING = {
+      initial: (
+        <ListGroup>
+          <ListGroupItem
+            className="text-decoration-none"
+            href="#"
+            onClick={() => this.setState({step: 'url'})}
+            tag="a"
+          >
+            <FontAwesomeIcon icon={faCode} /> URL
+          </ListGroupItem>
+          <ListGroupItem
+            className="text-decoration-none"
+            href="#"
+            onClick={() => this.setState({step: 'upload'})}
+            tag="a"
+          >
+            <FontAwesomeIcon icon={faUpload} /> Upload
+          </ListGroupItem>
+        </ListGroup>
+      ),
+      metadata: (
+        <ImageMetadataForm
+          attributes={this.state.attributes}
+          onChange={(name, value) => this.setAttribute(name, value)}
+          onSubmit={() => this.setState({step: 'renderingHints'})}
+        />
+      ),
+      renderingHints: (
+        <ImageRenderingHintsForm
+          attributes={this.state.attributes}
+          onChange={(name, value) => this.setAttribute(name, value)}
+          onSubmit={() => this.addImageToEditor()}
+        />
+      ),
+      upload: (
+        <ImageUploadForm
+          apiContextPath={this.props.apiContextPath}
+          onChange={attributes => this.setAttributes(attributes)}
+          onSubmit={() => this.setState({readOnly: true, step: 'url'})}
+        />
+      ),
+      url: (
+        <ImageUrlForm
+          attributes={this.state.attributes}
+          onChange={(name, value) => this.setAttribute(name, value)}
+          onSubmit={() => this.setState({step: 'metadata'})}
+          readOnly={this.state.readOnly}
+        />
+      ),
+    }
+    return STEP_COMPONENT_MAPPING[this.state.step]
+  }
+
+  getStepTitle = t => {
+    const STEP_TITLE_MAPPING = {
```

Comment:
Das sollte als echte Konstante in den globalen Scope

Reply:
Die Strings müssten auch noch internationalisiert werden

Reply:
Wenn die String erstmal internationalisiert sind, hängen sie aber auch vom Zustand ab --> wäre also auch keine Konstante mehr, oder?